learning/step_by_step/ui_code_a_life_bar.rst

```diff
@@ -0,0 +1,590 @@
+    Learn more about _ready: :doc:`step_by_step/scripting_continued`
+
+Update health with a signal when the player takes a hit
+-------------------------------------------------------
+
+Our GUI is ready to receive the ``health`` value updates from the
+``Player``. To achieve this we're going to use **signals**.
+
+.. note::
+
+::
+
+    There are many useful built-in signals like `enter_tree` and `exit_tree`, that all nodes emit when they are respectively created and destroyed. You can also create your own using the `signal` keyword. On the `Player` node, you'll find two signals we created for you: `died` and `took_damage`.
+
+Why don't we directly get the ``Player`` node in the ``_process``
+function and look at the health value? Accessing nodes this way creates
+tight coupling between them. If you dio it sparingly it may work. As
+your game grows bigger, you may have many more connections. If you get
+nodes from a bad it's becomes very complex very soon. Not only that: you
+need to listen to the changes state constantly in the ``_process``
+function. The check happens 60 times a second and you'll likely break
+the game because of the order in which the code runs.
+
+On a given frame you may look at another node's property *before* it was
+updated: you get a value that from the last frame. This leads to obscure
+bugs that are hard to fix. On the other hand, a signal is emitted right
+after a change happened. It **guarantees** you're getting a fresh piece
+of information. And you will update the state of your connected node
+*right after* the change happened.
+
+.. note::
+
+::
+
+    The Observer pattern, that signals derive from, still adds a bit of coupling between node branches. But it's generally lighter and more secure than accessing nodes directly to communicate between two separate classes. It can be okay for a parent node to get values from its children. But you'll want to favor signals if you're working with two separate branches.
+    Read Game Programming Patterns for more information on the [ Observer pattern ](http://gameprogrammingpatterns.com/observer.html).
+    The [ full book ](http://gameprogrammingpatterns.com/contents.html) is available online for free.
+
+With this in mind let's connect the ``GUI`` to the ``Player``. Click on
+the ``Player`` node in the scene dock to select it. Head down to the
+Inspector and click on the Node tab. This is the place to connect nodes
+to listen the one you selected.
+
+The first section lists custom signals defined in ``player.GD``:
+
+-  ``died`` is emitted when the character just died. We will use it in a
+   moment to hide the UI.
+-  ``took_damage`` is emitted when the character got hit.
+
+.. figure:: img/took_damage_signal.png
+   :alt: We're connecting to the took\_damage signal
+
+   We're connecting to the took\_damage signal
+
+Select ``took_damage`` and click on the Connect button in the bottom
```

Comment:
I'd rather name it "health_changed", since you would like to update the healthbar when the player is healed.

Reply:
Yeah, thanks for pointing this out. I actually changed it in a new demo I'm working on today (attack mechanic)